src/upgrade.py

```diff
@@ -101,6 +102,9 @@ def _on_upgrade_granted(self, event: UpgradeGrantedEvent) -> None:
         logger.info(f"{self.charm.unit.name} upgrading service...")
         self.charm.snap.restart_snap_service()
 
+        # Allow for some time to settle down
+        time.sleep(20)
+
```

Comment:
**todo:** I would mark this as `FIXME`, but it's probably fine for the short term.

I don't think we want to rely on a hard-coded value (any value really). We can use the `pause` upgrade-strategy that was mentioned recently in the newer upgrade specs for production deployments, which avoids this issue.

Or, we need to add more detailed 'ready' checks, and use them.

Reply:
Posting also here: https://warthogs.atlassian.net/browse/DPE-3155